src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditor.jsx

```diff
@@ -0,0 +1,304 @@
+import React from 'react';
+import { useIntl } from 'react-intl';
+import { get, filter } from 'lodash-es';
+
+import { FormattedMessage } from 'react-intl';
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import NewChildIcon from '@material-ui/icons/AddCircle';
+import TextInput from '../../../../components/inputs/TextInput';
+import DeleteButton from '../../../../components/DeleteButton';
+import Button from '../../../../components/Button';
+import Text from '../../../../components/Text';
+
+const intl = useIntl();
+
+function recomposeAllRelationshipObjectFromModifiedCategoryObject(
```

Comment:
i know names like this are very clear but at some point they are just too long lmao

Reply:
Supercalifragilisticexpialidocious is 34 characters and hereafter the upper limit on variable names. yours is 56!

Reply:
Done.